### ISSUE TITLE
Use Rust API directly from flate2

### ISSUE THREAD

**Issue by matklad** [Frommi/miniz_oxide#16]
An interesting problem was reported at https://gitter.im/rust-lang/rust?at=5a57cc7183152df26d5d5cfc.

Basically, `miniz-sys` and `miniz_oxide_c_api` expose the same C-API, so you can't link them both into the same process. So it might make sense to bypass C-API in flate2 and avoid polluting linker's namespace.

**Comment by oyvindln** [comment]
Yeah, that was the plan. Though a fix for now, there is a no_c_export feature in miniz_oxide_c_api, which among other things makes the exported functions mangled, so they should not conflict. Maybe this feature should be enabled by default, or added in the dependencies section of flate2.

**Comment by Frommi** [comment]
There seems to be an easy way to [choose features](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#choosing-features) of dependencies now that if I remember correctly wasn't there a few months ago. So it may be good idea to advertise `no_c_export` on main readme and add that to flate2.

**Comment by oyvindln** [comment]
Added a [PR](https://github.com/alexcrichton/flate2-rs/pull/145) to to flate2 to use the no_c_export feature. I wasn't able to trigger the error myself, so I can't guarantee that it will fix the problem, but hopefully it will.
